### PR TITLE
Rename option "idField" to "keyField".

### PR DESCRIPTION
**Pull request by ADmad**
The latter is more intuitive. There was a ticket for it too (can't find it).

### CODE REVIEW

**Comment by dereuromark** [comment]
I think this makes a lot of sense.

Will also need a migration note of course then.

**Comment by AD7six** [comment]
:+1:

File: src/ORM/Table.php [hunk 1]
```diff
@@ -943,18 +943,25 @@ public function findAll(Query $query, array $options)
...
-            'idField' => $this->primaryKey(),
+            'keyField' => $this->primaryKey(),
             'valueField' => $this->displayField(),
             'groupField' => null
         ];
+
+        if (isset($options['idField'])) {
+            $options['keyField'] = $options['idField'];
+            unset($options['idField']);
+            trigger_error('Option "idField" is deprecated, use "keyField" instead.', E_USER_WARNING);
```
Comment (markstory) [inline src/ORM/Table.php]: Thanks :+1:

**Comment by markstory** [comment]
Related RFC issue is #4832

**Comment by jhli** [comment]
Thanks!

What about using the hash key displayField instead of valueField see the linked RFC for arguments.

**Comment by AD7six** [comment]
I don't think displayField make sense.

In a list there is a key and a value; "display" only makes sense in certain use cases (I'm going to use this list to display the array values).

**Comment by jhli** [comment]
For the most common use - populating selects/radio it does but for a generic key-value list, it doesn't. K=V via this ticket is already a lot better than id=value. Thanks.

**Comment by caBBAlainB** [comment]
Just for info on the last update made this morning, but I suppose you are aware of.

Warning (512): Option "idField" is deprecated, use "keyField" instead. [CORE\src\ORM\Table.php, line 954]

**Comment by ADmad** [comment]
@caBBAlainB That warning is intentional to make users update their code to use new option name.

**Comment by caBBAlainB** [comment]
Oh yes, stupid!

**Pull request merged**